### PR TITLE
Ensure project id meets gcp requirements

### PR DESCRIPTION
**Pull request by NautiluX**
OSD-3007

### CODE REVIEW

**Comment by NautiluX** [comment]
Thanks for pointing this out. I added tests for all the requirements and changed to a different generation:
- generate v4 GUID
- sha1 hash GUID
- take first 26 bytes as characters
- add `osd-` prefix to ensure it starts with a letter

This should still have a sufficient low probability for collisions, while being compliant to the naming requirements from GCP.

File: pkg/controller/projectreference/projectreference_controller_test.go [hunk 1]
```diff
@@ -83,6 +83,30 @@ var _ = Describe("ProjectReference controller reconcilation", func() {
 			})
 		})
 
+		Context("generated project names", func() {
+			It("are shorter than 30 characters", func() {
+				projectName := generateProjectName()
+				Expect(len(projectName)).To(BeNumerically("<=", 30))
+			})
+
+			It("are shorter than 6 characters", func() {
```
Comment (drpaneas) [inline pkg/controller/projectreference/projectreference_controller_test.go]:
```suggestion
			It("are longer than 6 characters", func() {
```

**Pull request merged**